pym/bob/input.py

```diff
@@ -3476,7 +3507,18 @@ def __parse(self, envOverrides, platform, recipesRoot=""):
         self.__rootRecipe = Recipe.createVirtualRoot(self, sorted(filteredRoots), self.__properties)
         self.__addRecipe(self.__rootRecipe)
 
-    def __parseLayer(self, layer, maxVer, recipesRoot):
+    def updateLayers(self, loop, defines, verbose):
+        from .layers import fetchLayers
+        try:
+            self.parse(defines, dryRun=True)
+        except ParseError:
+            pass
+        self.__layers = []
+        fetchLayers(self, self.__cache, loop, verbose)
+
```

Comment:
That should be moved into the layers module. The input module should only deal with parsing the recipes.